internal/database/db_test.go

```diff
@@ -456,65 +513,67 @@ func TestMapFilesToDataset(t *testing.T) {
 		success := sqlmock.NewResult(1, 1)
 
 		for di, acs := range diSet {
+
+			mock.ExpectExec("INSERT INTO sda.datasets \\(stable_id\\) VALUES \\(\\$1\\) " +
+				"ON CONFLICT DO NOTHING;").
+				WithArgs(di).
+				WillReturnResult(success)
 			mock.ExpectBegin()
 			for _, aID := range acs {
-				r := sqlmock.NewRows([]string{"file_id"})
 
-				fileID, _ := strconv.Atoi(aID)
+				r := sqlmock.NewRows([]string{"id"})
+
+				// fileID, _ := strconv.Atoi(aID)
 
-				r.AddRow(fileID)
+				r.AddRow(aID)
 
-				mock.ExpectQuery("SELECT file_id FROM local_ega.archive_files WHERE stable_id = \\$1").
+				mock.ExpectQuery("SELECT id FROM sda.files WHERE stable_id = \\$1;").
 					WithArgs(aID).WillReturnRows(r)
 
-				mock.ExpectExec("INSERT INTO local_ega_ebi.filedataset "+
-					"\\(file_id, dataset_stable_id\\) VALUES \\(\\$1, \\$2\\) "+
-					"ON CONFLICT "+
-					"DO NOTHING;").
-					WithArgs(fileID, di).WillReturnResult(success)
+				mock.ExpectExec("INSERT INTO sda.file_dataset "+
+					"\\(file_id, dataset_id\\) SELECT \\$1, id FROM sda.datasets WHERE stable_id = \\$2 "+
+					"ON CONFLICT DO NOTHING;").
+					WithArgs(aID, di).WillReturnResult(success)
 
 			}
 			mock.ExpectCommit()
-
 			err := testDb.MapFilesToDataset(di, acs)
 
 			if !assert.Nil(t, err, "MapFilesToDataset failed unexpectedly") {
 				return err
 			}
 		}
 
-		var buf bytes.Buffer
-		log.SetOutput(&buf)
+		// var buf bytes.Buffer
```

Comment:
I'm not a huge fan of keeping commented out code. Would it be ok to remove it instead?

Reply:
i need to redo that test ... i missed it